### DISCUSSION TITLE
Rename Environment Variable for Access Token

### DISCUSSION THREAD

**Discussion by giannidallatorre** [tdviet/fedcloudclient#205] [Ideas]
## Suggestion: Rename Environment Variable for Access Token

I recommend renaming the environment variable for the access token from:

```
OIDC_ACCESS_TOKEN
```

to:

```
OIDC_AT
```

This change will harmonize the naming convention with the `oidc-agent` standards, as described in the [oidc-agent documentation](https://indigo-dc.gitbook.io/oidc-agent/user/oidc-token/options#token).

By making this adjustment, it will be easier to load the token for the `fedlcloud` client using the `-c` option of the `oidc` client. For example:

```sh
eval `oidc-token -c egi`
```

This will ensure a smoother and more consistent experience when working with the `oidc-agent`.
Thank you for considering this suggestion!

---

**Comment by tdviet** [comment]
Hi Gianni, thank you for useful suggestion. Keeping compatibilities with `oidc-agent` is really recommended.

Just for information, in upcoming version 2.0 (now in 2.0-alpha1 branch), we will introduce a unified naming scheme for all parameters/variables, see https://fedcloudclient.readthedocs.io/en/2.0-alpha1/usage.html#configuration. In short, a parameter; e.g. `site`, can be set by the command-line option with the same name; i.e. `--site`, the environment variable starting with the prefix `FEDCLOUD_`; i.e. `FEDCLOUD_SITE`. Therefore, the access token would be set in the `FEDCLOUD_ACCESS_TOKEN` environment variable.  This is the default behavior, all environment variables with prefix `FEDCLOUD_` will be automatically loaded to internal parameters with the same names, see https://github.com/tdviet/fedcloudclient/blob/2.0-alpha1/fedcloudclient/conf.py#L78.

However, we can add a "compatibility mode", an additional function with few lines of codes, to check if  `FEDCLOUD_ACCESS_TOKEN` is not set but `OIDC_AT` is, the access token will be read from the later variable. As long as we will have everything documented, it is clear and understandable.

**Comment by tdviet** [comment]
Feature request added for keeping track. See https://github.com/tdviet/fedcloudclient/issues/206

**Reply by giannidallatorre** [reply]
Thanks, Viet! I really appreciate it.
@enolfc also pointed out the OIDC_AGENT_ACCOUNT variable to me, which makes integrating with OIDC-AGENT a lot smoother. My bad for not having spotted that in the documentation earlier!
